src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -135,7 +135,7 @@ ss::future<> metadata_dissemination_service::start() {
       all_broker_addresses.begin(),
       all_broker_addresses.end());
 
-    (void)ss::with_gate(
+    ssx::spawn_with_gate(
```

Comment:
I wonder if there's any utility in having something like:
```c++
ssx::spawn_with_gate_invoke(
   bg,
   &metadata_dissemination_service::update_metadata_with_retries,
   this,
   std::move(addresses);
```
I guess that didn't turn out quite as tidy as I'd hoped.

Reply:
Mmm, I'm inclined to stick with lambdas mostly, although there are patterns like "argument-less member function that just runs in the background on a timer" that could benefit from another higher level wrapper in future.